tests/main.cpp

```diff
@@ -510,7 +511,9 @@ TEST_CASE( "WiringTest" )
 
     // Disconnect a component
 
+    circuit->PauseAutoTick();
     probe->DisconnectInput( 0 );
+    circuit->ResumeAutoTick();
```

Comment:
Not sure I understand why you added these Pause and Resume calls here?

Reply:
The pause and resume are to protect inputWires vector from getting modified from one thread while its being accessed from another. Tests crashed (only) sometimes because of this.

Note: A much better solution would be to add a lock around inputWires but this would have an impact on performance.
Note: Disconnecting input using circuit is not a problem because you have a pause/ resume in DisconnectComponent method.

Reply:
Oh I see. This is good stuff @linux-admirer, thank you so much for taking the time to write this!